RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitProducers.cc

```diff
@@ -0,0 +1,147 @@
+    desc.add("configSource", edm::ESInputTag{})->setComment("Label for ROC configuration parameters");
+    desc.add<int>("n_blocks", -1);
+    desc.add<int>("n_threads", -1);
+    desc.add<int>("n_hits_scale", -1);
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  void HGCalRecHitsProducer::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
+    auto queue = iEvent.queue();
+
+    // Read digis
+    auto const& deviceCalibParamProvider = iSetup.getData(calibToken_);
+    auto const& deviceConfigParamProvider = iSetup.getData(configToken_);
+    auto const& hostDigisIn = iEvent.get(digisToken_);
+
+    //printout new conditions if available
+    LogDebug("HGCalCalibrationParameter").log([&](auto& log) {
+      if (calibWatcher_.check(iSetup)) {
+        for (int i = 0; i < deviceConfigParamProvider.view().metadata().size(); i++) {
+          log << "gain = " << deviceConfigParamProvider.view()[i].gain() << "\n";
+        }
+      }
+    });
+    LogDebug("HGCalCalibrationParameter").log([&](auto& log) {
+      if (calibWatcher_.check(iSetup)) {
+        for (int i = 0; i < deviceCalibParamProvider.view().metadata().size(); i++) {
+          log << "idx = " << i << ", "
+              << "ADC_ped = " << deviceCalibParamProvider.view()[i].ADC_ped() << ", "
+              << "CM_slope = " << deviceCalibParamProvider.view()[i].CM_slope() << ", "
+              << "CM_ped = " << deviceCalibParamProvider.view()[i].CM_ped() << ", "
+              << "BXm1_slope = " << deviceCalibParamProvider.view()[i].BXm1_slope() << ", " << std::endl;
+        }
+      }
+    });
+
+    int oldSize = hostDigisIn.view().metadata().size();
+    int newSize = oldSize * n_hits_scale;
+    auto hostDigis = HGCalDigiHost(newSize, queue);
+    // TODO: replace with memcp ?
+    for (int i = 0; i < newSize; i++) {
+      //hostDigis.view()[i].electronicsId() = hostDigisIn.view()[i%oldSize].electronicsId();
+      hostDigis.view()[i].tctp() = hostDigisIn.view()[i % oldSize].tctp();
+      hostDigis.view()[i].adcm1() = hostDigisIn.view()[i % oldSize].adcm1();
+      hostDigis.view()[i].adc() = hostDigisIn.view()[i % oldSize].adc();
+      hostDigis.view()[i].tot() = hostDigisIn.view()[i % oldSize].tot();
+      hostDigis.view()[i].toa() = hostDigisIn.view()[i % oldSize].toa();
+      hostDigis.view()[i].cm() = hostDigisIn.view()[i % oldSize].cm();
+      hostDigis.view()[i].flags() = hostDigisIn.view()[i % oldSize].flags();
+      //LogDebug("HGCalCalibrationParameter")
+      //  << "idx=" << i << ", elecId=" << hostDigis.view()[i].electronicsId()
+      //  << ", cm=" << hostDigis.view()[i].cm() << std::endl;
+    }
+    LogDebug("HGCalRecHitsProducer") << "Loaded host digis: " << hostDigis.view().metadata().size();  //<< std::endl;
+
+    LogDebug("HGCalRecHitsProducer") << "\n\nINFO -- calling calibrate method";  //<< std::endl;
+
+#ifdef EDM_ML_DEBUG
+    alpaka::wait(queue);
+    auto start = std::chrono::high_resolution_clock::now();
```

Comment:
By the way, for measuring durations the `std::chrono::steady_clock` is recommended because it can go only forward. The time in `std::chrono::high_resolution_clock` is allowed to be adjusted at any time.

Reply:
this substitution had been overlooked - fixed